aerospike/check.py

```diff
@@ -1,77 +1,205 @@
+        # ns=phobos_sindex:set=longevity:indexname=str_uniq_idx:num_bins=1:bins=str_uniq_bin:type=TEXT:sync_state=synced:state=RW; \
+        # ns=phobos_sindex:set=longevity:indexname=int_uniq_idx:num_bins=1:bins=int_uniq_bin:type=INT SIGNED:sync_state=synced:state=RW;
 
+        # $ asinfo -v 'sets/bar'
+        # ns=bar:set=demo:objects=1:tombstones=0:memory_data_bytes=34:truncate_lut=0:stop-writes-count=0:set-enable-xdr=use-default:disable-eviction=false;\
+        # ns=bar:set=demo2:objects=123456:tombstones=0:memory_data_bytes=8518464:truncate_lut=0:stop-writes-count=0:set-enable-xdr=use-default:disable-eviction=false
+
+        match = re.match('^ns=%s:([^:]+:)?%s=([^:]+):.*$' % (namespace, secondary), l)
+        if match == None:
+            continue
+        idxs.append(match.groups()[1])
+
+    return idxs   
 
 class AerospikeCheck(AgentCheck):
 
     def __init__(self, name, init_config, agentConfig, instances=None):
         AgentCheck.__init__(self, name, init_config, agentConfig, instances)
-        self.connection_pool = {}
+        self.connections = {}
 
     def check(self, instance):
-        """Run the Aerospike check for one instance."""
+        addr, metrics, namespace_metrics, required_namespaces, tags = \
+            self._get_config(instance)
 
-        # Required parameters
-        host = instance['host']
-        port = int(instance['port'])
+        try:
+            conn = self._get_connection(addr)
 
-        # Optional parameters
-        want_metrics = set(instance.get('metrics', []))
-        want_namespace_metrics = set(instance.get('namespace_metrics', []))
-        namespaces = instance.get('namespaces', None)
+            with closing(conn.makefile('r')) as fp:
+                conn.send('statistics\r')
+                self._process_data(fp, CLUSTER_METRIC_TYPE, metrics, tags=tags)
+
+                namespaces = self._get_namespaces(conn, fp, required_namespaces)
+
+                for ns in namespaces:
+                    conn.send('namespace/%s\r' % ns)
+                    self._process_data(fp, NAMESPACE_METRIC_TYPE, namespace_metrics, tags+['namespace:%s' % ns])
+
+                    conn.send('sindex/%s\r' % ns)
+                    for idx in parse_namespace(fp.readline().split(';')[:-1], ns,'indexname'):
+                        conn.send('sindex/%s/%s\r' % (ns,idx))
+                        self._process_data(fp, SINDEX_METRIC_TYPE, [], 
+                                tags+['namespace:%s' % ns, 'sindex:%s.%s' % (ns,idx)])
+                    
+                    conn.send('sets/%s\r' % ns)
+                    for s in parse_namespace(fp.readline().split(';'),ns,'set'):
+                        conn.send('sets/%s/%s\r' % (ns,s))
+                        self._process_data(fp, SET_METRIC_TYPE, [],
+                                tags+['namespace:%s' % ns, 'set:%s.%s' % (ns,s)], ':')
```

Comment:
ditto to the comment for `sindex` regarding the `set` tag. Also, again, what might be the expected order of magnitude for `sets`?

Reply:
1023 sets per namespace

Reply:
This is a bit high... I know this is the upper bound. Let me get back to you regarding this.